include/openssl/kdf.h

```diff
@@ -90,6 +93,19 @@ extern "C" {
             EVP_PKEY_CTX_ctrl_uint64(pctx, -1, EVP_PKEY_OP_DERIVE, \
                             EVP_PKEY_CTRL_SCRYPT_MAXMEM_BYTES, maxmem_bytes)
 
+# define EVP_PKEY_CTX_set_pbkdf2_md(pctx, md) \
+            EVP_PKEY_CTX_ctrl(pctx, -1, EVP_PKEY_OP_DERIVE, \
+                              EVP_PKEY_CTRL_PBKDF2_MD, 0, (void *)(md))
+
+# define EVP_PKEY_CTX_set1_pbkdf2_salt(pctx, salt, saltlen) \
```

Comment:
Very cool, I like the suggestion and just created a PR for it: https://github.com/openssl/openssl/pull/4197 -- in order to fully test all KDFs (old and new-style getters), I've also created a second PR https://github.com/openssl/openssl/pull/4196 that adds missing tests -- but you already know this since you've already approved it :-)